cmake/common.cmake

```diff
@@ -1,660 +1,679 @@
+            list(APPEND _components_variant_doxysphinx_targets ${component_name}_doxysphinx)
+        endif()
+    endforeach()
+
+    set(_cov_out_variant_html reports/html/${_rel_reports_output_dir}/coverage/index.html)
+    add_custom_command(
+        OUTPUT ${_cov_out_variant_html}
+        COMMAND gcovr --root ${CMAKE_SOURCE_DIR} --add-tracefile \"${CMAKE_CURRENT_BINARY_DIR}/**/${COV_OUT_JSON}\" --html --html-details --output ${_cov_out_variant_html}
+        DEPENDS ${GLOBAL_COMPONENTS_COVERAGE_JSON_LIST}
+        COMMENT "Generating overall code coverage report ${_cov_out_variant_html} ..."
+    )
+
+    add_custom_target(
+        _components_variant_coverage_html_target
+        DEPENDS ${_components_coverage_html} ${_cov_out_variant_html}
+    )
+
+    # add the generated files as dependency to cmake configure step
+    set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${_reports_config_json})
+    add_custom_target(
+        reports
+        ALL
+        COMMAND ${CMAKE_COMMAND} -E make_directory ${_reports_output_dir}
+
+        # We need to call sphinx-build with -E to make sure all files are regenerated.
+        COMMAND ${CMAKE_COMMAND} -E env SPHINX_BUILD_CONFIGURATION_FILE=${_reports_config_json} AUTOCONF_JSON_FILE=${AUTOCONF_JSON} VARIANT=${VARIANT} -- sphinx-build -E -b html ${PROJECT_SOURCE_DIR} ${_reports_html_output_dir}
+        BYPRODUCTS ${_reports_html_output_dir}/index.html
+        DEPENDS _components_variant_coverage_html_target ${_components_variant_doxysphinx_targets}
+    )
+endmacro()
+
+macro(_spl_set_coverage_create_overall_report_is_necessary)
+    set(_SPL_COVERAGE_CREATE_OVERALL_REPORT_IS_NECESSARY TRUE PARENT_SCOPE)
+endmacro(_spl_set_coverage_create_overall_report_is_necessary)
+
+set(COV_OUT_JSON coverage.json)
+
+function(_spl_coverage_create_overall_report)
+    if(_SPL_COVERAGE_CREATE_OVERALL_REPORT_IS_NECESSARY)
+        set(COV_OUT_VARIANT_HTML reports/coverage/index.html)
+        add_custom_command(
+            OUTPUT ${COV_OUT_VARIANT_HTML}
+            COMMAND gcovr --root ${CMAKE_SOURCE_DIR} --add-tracefile \"${CMAKE_CURRENT_BINARY_DIR}/**/${COV_OUT_JSON}\" --html --html-details --output ${COV_OUT_VARIANT_HTML}
+            DEPENDS ${GLOBAL_COMPONENTS_COVERAGE_JSON_LIST}
+            COMMENT "Generating overall code coverage report ${COV_OUT_VARIANT_HTML} ..."
+        )
+        add_custom_target(
+            unittests
+            DEPENDS coverage ${COV_OUT_VARIANT_HTML}
+        )
+        add_custom_target(
+            coverage_overall_report
+            DEPENDS ${COV_OUT_VARIANT_HTML}
+        )
+    else(_SPL_COVERAGE_CREATE_OVERALL_REPORT_IS_NECESSARY)
+        add_custom_target(unittests)
+    endif(_SPL_COVERAGE_CREATE_OVERALL_REPORT_IS_NECESSARY)
+endfunction(_spl_coverage_create_overall_report)
+
+macro(_spl_add_test_suite COMPONENT_NAME PROD_SRC TEST_SOURCES)
+    _spl_set_coverage_create_overall_report_is_necessary()
+
+    set(exe_name ${COMPONENT_NAME}_test)
```

Comment:
Really, `component_name` was a global variable only used by this macro? Nice catch 👍

Reply:
Not really, create_component() is calling _spl_add_test_suite with some args. But not component_name which is the important one. So I just defined is as parameter to make the dependencies more clear. But yes, a nice catch nevertheless. 😅